File: Interview_Prep_App/native_java/javatemplate/src/main/java/com/tc2r/javatemplate/repositories/tc2rgithubrepository/source/remote/Tc2rGithubService.java
package com.tc2r.javatemplate.repositories.tc2rgithubrepository.source.remote;


import com.tc2r.javatemplate.repositories.models.Answer;

import java.util.ArrayList;
import java.util.Arrays;

public interface Tc2rGithubService {
    // Request method and URL specified in the annotation

    // FIXME: Use your own preference to make request to the api
    default ArrayList<Answer> getAnswers() {
        Answer tempAnswer = new Answer("It will perform the inter connection between activities and the data " + "passing mechanism.",
                                       "An Intent is connected to either the external world of application or " + "internal world of application ,Such as, opening a pdf is an intent and connect to the web browser" + ".etc.");

        return new ArrayList<>(Arrays.asList(tempAnswer, tempAnswer, tempAnswer, tempAnswer, tempAnswer, tempAnswer));
    }

    // FIXME: Use your own preference to make request to the api
    void getQuestionsList();
}

File: Interview_Prep_App/native_kotlin/template/src/main/java/com/interviewprep/template/repository/tc2rgithubrepository/source/remote/Tc2rGithubService.kt
package com.interviewprep.template.repository.tc2rgithubrepository.source.remote

import com.interviewprep.template.repository.models.response.AnswersResponse
import com.interviewprep.template.repository.models.response.QuestionsResponse

internal interface Tc2rGithubService {
    // Request method and URL specified in the annotation

    // FIXME: Use your own preference to make request to the api
    fun getAnswers(onAnswersReceived: (AnswersResponse?, Throwable?) -> Unit) {
        onAnswersReceived(AnswersResponse(), null)
    }

    // FIXME: Gets answers objects
    fun getQuestions(onQuestionsReceived: (QuestionsResponse?, Throwable?) -> Unit) {
        onQuestionsReceived(QuestionsResponse(), null)
    }

    // Add any other request you'd like app to make as well.
}
